hack/release.go

```diff
@@ -0,0 +1,301 @@
+		fmt.Printf("Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	for _, file := range files {
+		tmpl, err := template.New(filepath.Base(file)).Funcs(template.FuncMap{
+			"imageAsSha": func(obj interface{}) string {
+				imageSpec := obj.(*ReleaseImageSpec)
+				return imageSpec.Shas[0].ImageRef
+			},
+		}).ParseFiles(file)
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
+			os.Exit(1)
+		}
+
+		// Generate new file full path
+		outputFile := filepath.Join(*outputDir, strings.Replace(filepath.Base(file), ".template", ".yaml", 1))
+		f, err := os.Create(filepath.Clean(outputFile))
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
+			os.Exit(1)
+		}
+		err = tmpl.Execute(f, release)
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
+			os.Exit(1)
+		}
+	}
+}
+
+func resolveImagesSha(release *Release) error {
+	nvcrToken := os.Getenv("NGC_CLI_API_KEY")
+	if nvcrToken == "" {
+		return fmt.Errorf("NGC_CLI_API_KEY is unset")
+	}
+	auth := &authn.Basic{
+		Username: "$oauthtoken",
+		Password: nvcrToken,
+	}
+	v := reflect.ValueOf(*release)
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i)
+		if !field.IsNil() {
+			releaseImageSpec := field.Interface().(*ReleaseImageSpec)
+			if strings.Contains(releaseImageSpec.Image, "doca-driver") {
+				digests, err := resolveDocaDriversShas(releaseImageSpec.Repository, releaseImageSpec.Image,
+					releaseImageSpec.Version, auth)
+				if err != nil {
+					return err
+				}
+				releaseImageSpec.Shas = make([]SHA256ImageRef, len(digests))
+				for i, digest := range digests {
+					sha := fmt.Sprintf("%s/%s@%s", releaseImageSpec.Repository, releaseImageSpec.Image, digest)
+					releaseImageSpec.Shas[i] = SHA256ImageRef{ImageRef: sha, Name: fmt.Sprintf("doca-driver-%d", i)}
+				}
+			} else {
+				digest, err := resolveImageSha(releaseImageSpec.Repository, releaseImageSpec.Image,
+					releaseImageSpec.Version, auth)
+				if err != nil {
+					return err
+				}
+				releaseImageSpec.Shas = make([]SHA256ImageRef, 1)
+				sha := fmt.Sprintf("%s/%s@%s", releaseImageSpec.Repository, releaseImageSpec.Image, digest)
+				releaseImageSpec.Shas[0] = SHA256ImageRef{ImageRef: sha}
+			}
+		}
+	}
+	return nil
+}
+
+func resolveImageSha(repo, image, tag string, auth *authn.Basic) (string, error) {
+	ref, err := containerregistryname.ParseReference(fmt.Sprintf("%s/%s:%s", repo, image, tag))
+	if err != nil {
+		return "", err
+	}
+	var desc *remote.Descriptor
+	if strings.Contains(repo, "nvstaging") {
+		desc, err = remote.Get(ref, remote.WithAuth(auth))
+		if err != nil {
+			return "", err
+		}
+	} else {
+		// Container registry might fail if providing unneeded auth
+		desc, err = remote.Get(ref)
+		if err != nil {
+			return "", err
+		}
+	}
+	digest, err := containerregistryv1.NewHash(desc.Descriptor.Digest.String())
+	if err != nil {
+		return "", err
+	}
+	return digest.String(), nil
+}
+
+func resolveDocaDriversShas(repoName, imageName, ver string, auth *authn.Basic) ([]string, error) {
```

Comment:
do we need this one ?

Reply:
No, but I prefer to have the `release.go` same as in network operator, so that if there is an update, it will be a simple replace

Reply:
i was thinking that we may diverge the two each would keep only what it needs.
so that if we require an update which is relevant only for docs we modify release.go in this repo.

Reply:
Done